### ISSUE TITLE
Corrupted VCF headers

### ISSUE THREAD

**Issue by wdecoster** [HKU-BAL/Clair3#25]
Hi,

I am very much impressed by the speed and accuracy of Clair3! But I noticed that for some of my merge_output.vcf.gz output files the header line gets corrupted. I only have this error after adding `--vcf_fn=input_vars.vcf.gz`. 

I ran Clair3 like:
`bash /<path>/run_clair3.sh --bam_fn={} --vcf_fn=input_vars.vcf.gz --print_ref_calls --ref_fn=GRCh38.fa --threads=6  --platform="ont" --model_path="/<path>/Clair3/models/ont" --output=force_calls/{.} --haploid_sensitive --ctg_name=<a_specific_chromosome>`

I ran this in parallel for 12 bam files at a time, for >300 individuals

A few examples, in which I have to redact the chromosome name and variant positions...

The #CHROM line is not the last one in the header, and ##fileformat gets appended to the end of the next line. The ##fileformat line also happens as expected at the top of the file. Also note that chrUn_KI270372v1 is mentioned twice in the header, once correctly and once as shown below.
```
##contig=<ID=chrUn_GL000218v1,length=161147>
##contig=<ID=chrEBV,length=171823>
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  EMPTY
##contig=<ID=chrUn_KI270372v1,length=1650>##fileformat=VCFv4.2
```

The #CHROM line is not the last one in the header, and the first variant gets appended to the next line. Note that the first variant is diploid (0/1) while all other positions in the file are haploid (as they should be with  --haploid_sensitive). Also note that chrUn_KI270372v1 is mentioned twice in the header, once correctly and once as shown below
```
##contig=<ID=chrUn_GL000216v2,length=176608>
##contig=<ID=chrUn_GL000218v1,length=161147>
##contig=<ID=chrEBV,length=171823>
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  EMPTY
##contig=<ID=chrUn_KI270372v1,length=1650>chr<N> 123456        .       T       C       3.46    PASS    P       GT:GQ:DP:AF     0/1:3:1:1.0000
```

```
##contig=<ID=chrUn_GL000216v2,length=176608>
##contig=<ID=chrUn_GL000218v1,length=161147>
##contig=<ID=chrEBV,length=171823>
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  EMPTY
##contig=<ID=chrUn_KI270372v1,length=1650>##fileformat=VCFv4.2
##contig=<ID=chrUn_KI270372v1,length=1650>chr<N> 234567        .       T       T       21.02   RefCall P       GT:GQ:DP:AF     0/0:21:1:1.0000
```

Here ##fileformat=VCFv4.2 ended up in the middle of some variant records
```
chrA	pos1	.	T	C	6.97	PASS	F	GT:GQ:DP:AF	1:6:4:0.2500
chrA	pos2	.	C	C	20.33	RefCall	P	GT:GQ:DP:AF	0:20:4:1.0000
chrA	pos3	.	G	G	21.90	RefCall	F	GT:GQ:DP:AF	0:21:4:0.7500
chrA	pos4	.	T	T	23.93	RefCall	P	GT:GQ:DP:AF	0:23:4:1.0000
chrA	pos5	.	G	G	13.50	RefCall	F	GT:GQ:DP:AF	0:13:4:0.7500
chrA	pos6	.	A	A	26.66	RefCall	P	GT:GQ:DP:AF	0:26:4:1.0000##fileformat=VCFv4.2
chrA	pos7	.	G	G	18.71	RefCall	P	GT:GQ:DP:AF	0:18:4:0.7500
chrA	pos8	.	G	G	19.56	RefCall	P	GT:GQ:DP:AF	0:19:4:1.0000
```

Cheers,
Wouter

**Comment by aquaskyline** [comment]
Hi Wouter,

Thanks for trying out Clair3. We noticed that you said you ran 12 bam files at a time, not sure if some of the files got overwritten during the process. I noticed that you've masked the chromosomes and positions, but would it be possible for you to send us the log files (with sensitive filenames also masked) at my email address `rbluo@cs.hku.hk`? That helps us to pinpoint the problem way faster. Much appreciated!

**Comment by wdecoster** [comment]
Sure, do you mean the run_clair3.log files or the entire log directory created for each sample?

**Comment by aquaskyline** [comment]
All `${OUTPUT_DIR}/run_clair3.log` and `${OUTPUT_DIR}/log` if possible.

**Comment by aquaskyline** [comment]
It seems to us that the problem could be solved by using v0.1-r3 (released yesterday) and lowering the number of parallel jobs. Insufficient "maximum user processes" (check `ulimit -u`) might be the reason, a detailed description is in #24. Some jobs failed due to the "maximum user processes" has reached and generate some truncate VCF files. While the current implementation of the VCF merger doesn't check the integrity of the VCF file inputs, they were incorrectly merged.

We will add VCF input integrity check to the VCF merger in the next release.

**Comment by wdecoster** [comment]
Thanks for the quick fix. Great, I will try that!
